.github/workflows/build.yml

```diff
@@ -0,0 +1,58 @@
+name: panet-build
+
+on:
+  push:
+    branches: [ '*' ]
+    tags: [ 'v[0-9]+.[0-9]+.[0-9]+' ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: gitlab.desy.de:5555/paul.millar/panet-build:v0.4
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Needed to identify non-tagged builds
+
+      - name: Work-around for broken support for safe.directory
+        run: /usr/bin/git config --global --add safe.directory `pwd`
+
+      - name: Build PaNET and documentation
+        run: /bin/build-panet
+
+      - name: Archive generated PaNET
+        uses: actions/upload-artifact@v4
+        with:
+          name: PaNET
+          path: source/PaNET.owl
+          retention-days: 7
```

Comment:
If I understand well the OWL file will stay in github for 7 days. Where can users get the ontology after that?

Reply:
Here's my take on this.

You're right, the artefacts are only retained for 7 days.  The default is only one day.  Seven days is something of an arbitrary choice, but (to me) it felt about right.

These are only artefacts from the CI/CD.  Artefacts' primary use is to carrying information/state between CI/CD jobs.  If you want them longer-term then they should be "pushed" somewhere.

An important secondary use is to help developers double-check what they're doing.  For example, someone might pushing changes into a private branch and checking PaNET still builds and their change had the desired result on the ontology, before creating a pull request.

Finding the artefacts is surprisingly hard!

Here's one way to do it.  Click on the little green tick (or red cross), part of the one-line summary of a commit.  A pop-up window appears, listing the two jobs (`panet-build / build` and `panet-build / deploy`).  Select the `Details` link, on the right edge of the `build` job (`panet-build / build`).  Then select the "house-icon Summary" link.  You will find the artefacts towards the bottom of the page.

This UX is just so awful.  I'm hoping I've missed a more obvious way of navigating there, but if there is a better way then I haven't found it.

You try this out on [this branch](https://github.com/paulmillar/ExPaNDS-experimental-techniques-ontology/tree/add-builder-image) (in my fork of PaNET).

The artefacts should be deterministic.  If, after 7 days (and the artefacts have been purged) it turns out the information is useful then we _should_ be able to rerun the CI/CD workflow (or jobs therein) to regenerate them.

"Normal people" should obtain PaNET from one of GitHub releases, and not from the CI/CD artefacts.  These are independent and don't expire.

Where this 7 days expiry _does_ matter is when we're doing a release.  This isn't automated (at least, "not yet" :-).  When somebody tags a branch ("v1.5.0", say) then someone would have to download the PaNET artefacts to create the corresponding GitHub release.  That person would do download the artefact within 7 days.

Reply:
Terrible UX as you said or I am too much used to gitlab. Thanks for the clarification.